### PR TITLE
Update lightgbm nuget version so that it does not depend on gcc. [WIP]

### LINKED ISSUE #494: Not able to use LightGBM in .NET Framework 4.6.1
### System information

- **OS version/distro**:
Windows 10
- **.NET Version (eg., dotnet --info)**: 
.NET Framework 4.6.1
### Issue

- **What did you do?**
I am using the 0.3.0 version NuGet package.
I tried to replicate the example at https://www.microsoft.com/net/learn/apps/machine-learning-and-ai/ml-dotnet/get-started/windows, with the small change of replacing the classifier with a LightGbmClassifier. I was able to do this with a .NET Core application (after I understood I had to add the Microsoft.ML.LightGBM Nuget package). But I need to run this in a full .NET framework application. So I followed the same procedure for a .NET Framework 4.6.1 console application. But this failed when I tried to add the Microsoft.ML.LightGBM Nuget package.

- **What happened?**
Microsoft.ML.LightGBM depends on 'LightGBM 2.1.2.2', and I get the following error:

"Could not install package 'LightGBM 2.1.2.2'. You are trying to install this package into a project that targets '.NETFramework,Version=v4.6.1', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author."

- **What did you expect?**
I expected this to work, ML.NET generally works on .NET Framework 4.6.1, right?

### LINKED ISSUE #1067: Update LightGBM nuget so that there is no dependency on GCC
Currently lightgbm nuget depends on a binary for openmp that comes from gcc compiler but having to install gcc compiler on build machines causes slowdown. Update the nuget to the newest version that removes this dependency because lightgbm native code is compiled using clang.

### PR DESCRIPTION
**Pull request by codemzs**
fixes #1067
fixes #494

### CODE REVIEW

**Comment by Ivanidzo4ka** [comment]
Do we still need https://github.com/dotnet/machinelearning/pull/731 ?

**Comment by codemzs** [comment]
@Ivanidzo4ka I have closed that PR.

**Comment by eerhardt** [comment]
This seems to be broken on both linux and macOS. Do we know why @codemzs ?

File: src/Microsoft.ML.LightGBM/WrappedLightGbmInterface.cs [hunk 1]
```diff
@@ -150,6 +150,7 @@ public enum CApiDType : int
 
         [DllImport(DllName, EntryPoint = "LGBM_BoosterSaveModelToString", CallingConvention = CallingConvention.StdCall)]
         public static extern unsafe int BoosterSaveModelToString(IntPtr handle,
+            int startIteration,
```
Comment (eerhardt) [inline src/Microsoft.ML.LightGBM/WrappedLightGbmInterface.cs]: @guolinke -  what do you think about starting to follow https://semver.org/ with the LightGBM package? Check that site out and let me know what you think.

In general, it is surprising that there are API breaking changes in a minor update from `2.1.2.2` => `2.2.1`. Semantic versioning attempts to address these types of surprises. It really helps API customers out IMO.

File: test/Microsoft.ML.Core.Tests/Microsoft.ML.Core.Tests.csproj [hunk 2]
```diff
@@ -21,6 +21,7 @@
     <ProjectReference Include="..\..\src\Microsoft.ML.Legacy\Microsoft.ML.Legacy.csproj" />
     <ProjectReference Include="..\Microsoft.ML.TestFramework\Microsoft.ML.TestFramework.csproj" />
     <ProjectReference Include="..\Microsoft.ML.Tests\Microsoft.ML.Tests.csproj" />
+	<PackageReference Include="LightGBM" Version="$(LightGBMPackageVersion)" />
```
Comment (eerhardt) [inline test/Microsoft.ML.Core.Tests/Microsoft.ML.Core.Tests.csproj]: These test changes should not be necessary.

File: src/Microsoft.ML.LightGBM/WrappedLightGbmInterface.cs (hunk 1 above)
Comment (guolinke) [reply src/Microsoft.ML.LightGBM/WrappedLightGbmInterface.cs]: @eerhardt actually, we just released 2.2.0. 
2.2.1 is the special version for ML.Net

Comment (eerhardt) [reply src/Microsoft.ML.LightGBM/WrappedLightGbmInterface.cs]: Right - and in between `2.1.2.2` => `2.2.1` you made a public API breaking change.  See this change here.

Following semantic versioning, you cannot make a public API breaking change unless it is across major versions.

**Comment by montebhoover** [comment]
Sorry, I made a clumsy attempt to contribute here.  The particular way I merged master into this feature branch caused the last 30 commits to master to show up as changes in this PR.  Reverting it cleaned things back up, but the history remains.

The good news is that it builds successfully now...

File: docs/building/unix-instructions.md [hunk 3]
```diff
@@ -28,7 +28,7 @@ The following components are needed:
 * curl
 * All the requirements necessary to run .NET Core 2.0 applications: libssl1.0.0 (1.0.2 for Debian 9) and libicu5x (libicu52 for ubuntu 14.x, libicu55 for ubuntu 16.x, and libicu57 for ubuntu 17.x). For more information on prerequisites in different linux distributions click [here](https://docs.microsoft.com/en-us/dotnet/core/linux-prerequisites?tabs=netcore2x).
 
-For example, for Ubuntu 16.x:
+e.g. for Ubuntu 16.x:
```
Comment (montebhoover) [inline docs/building/unix-instructions.md]: This from a bad merge I did. Marking WIP until it is fixed.

**Comment by montebhoover** [comment]
My apologies to all, I made an incorrect merge on this branch.  I'm opening up a new pull request with the same changes here: #1234

**Pull request closed without merging**